index.xml

```diff
@@ -14374,6 +14528,155 @@ C implements A;</x:codeblock>
           </ol>
         </div>
 
+        <div id='es-namespaces' class='section'>
+          <h3>Namespaces</h3>
+
+          <p>
+            For every <a class='dfnref' href='#dfn-namespace'>namespace</a> that is <a
+            class='dfnref' href='#dfn-exposed'>exposed</a> in a given ECMAScript global environment,
+            a corresponding property <span class='rfc2119'>MUST</span> exist on the ECMAScript
+            environment's global object. The name of the property is the <a class='dfnref'
+            href='#dfn-identifier'>identifier</a> of the namespace, and its value is an object
+            called the <dfn data-export='' id='dfn-namespace-object'>namespace object</dfn>.
+          </p>
+          <p>
+            The property has the attributes <span class='descriptor'>{ [[Writable]]: <span
+            class='esvalue'>true</span>, [[Enumerable]]: <span class='esvalue'>false</span>,
+            [[Configurable]]: <span class='esvalue'>true</span> }</span>. The characteristics of a
+            namespace object are described in <a href='#namespace-object'>section <?sref
+            namespace-object?></a> <?sdir namespace-object?>.
+          </p>
+
+          <div id='namespace-object' class='section'>
+            <h4>Namespace object</h4>
+
+            <p>
+              The namespace object for a given <a class='dfnref' href='#dfn-namespace'>namespace</a>
+              <var>namespace</var> and Realm <var>realm</var> is created as follows:
+            </p>
+
+            <ol class='algorithm'>
+              <li>
+                Let <var>namespaceObject</var> be
+                <a>!</a> <a>ObjectCreate</a>(the <a>%ObjectPrototype%</a> of <var>realm</var>).
+              </li>
+
+              <li>
+                For each <a class='dfnref' href='#dfn-exposed'>exposed</a> <a class='dfnref'
+                href='#dfn-regular-operation'>regular operation</a> <var>op</var> that is a <a
+                class='dfnref' href='#dfn-namespace-member'>namespace member</a> of this namespace,
+
+                <ol>
+                  <li>
+                    Let <var>id</var> be <var>op</var>'s <a class='dfnref'
+                    href='#dfn-identifier'>identifier</a>.
+                  </li>
+
+                  <li>
+                    Let <var>steps</var> be the following series of steps, given function argument
```

Comment:
This was somewhat of an attempt to modernize that section to do things like state that methods are built-in functions (#106) and give property descriptors for name and length, and state the realm of the function. It also moves us toward a world where instead of stating what operations MUST be, we give an explicit procedure for creating interface objects and namespace objects. As I said in the OP, "The ES binding for namespaces here is written in a fully modern style, and so differs slightly from similar prose for interfaces' ES binding. It is hoped that it can provide a template for eventually updating interfaces' ES binding to modern ES."

Let me try factoring out this modern way of creating operations and replacing much of the existing #es-operations section with it. Then we can call into it from namespaces, and in the future call into it from interface objects.